File: palindrome_number/palindrome_number.c
#include "../leetcode.h"

bool isPalindrome(int x)
{
	if (x < 0)
		return 0;

	int i = x, rev = x % 10;

	while (i /= 10)
		rev = rev * 10 + i % 10;
	return x == rev;
}

int main()
{
	int n[] = {123, 121, -121, 10, -101, 1234567899};
	size_t i;

	for (i = 0; i < sizeof(n) / sizeof(*n); ++i)
		printf("%*i %s pallindrome\n", 8, n[i], isPalindrome(n[i]) == 1 ? "is" : "isn't");
	return 0;
}

File: palindrome_number/palindrome_number.go
package main

import "fmt"

func isPalindrome(x int) bool {
	if x < 0 {
		return false
	}

	var i, rev int = x / 10, x % 10

	for i > 0 {
		rev = rev * 10 + i % 10
		i /= 10
	}
	return x == rev
}

func main () {
	var x = []int{123, 121, -121, 10, -101, 1234567899}

	for _, v := range x {
		if isPalindrome(v) == true {
			fmt.Printf("%*d is a pallindrome\n", 8, v)
		} else {
			fmt.Printf("%*d is not a pallindrome\n", 8, v)
		}
	}
}
